src/services/side-panels/side-panels-manager.tsx

```diff
@@ -0,0 +1,171 @@
+    const item: ItemMetadata | undefined = this.componentsRegistry.get(itemId);
+    if (item) {
+      this.deactivateItem(itemId);
+      item.removeComponentFunc();
+      this.componentsRegistry.delete(itemId);
+    }
+  }
+
+  public activateItem(itemId: number): void {
+    const item: ItemMetadata | undefined = this.componentsRegistry.get(itemId);
+    if (item) {
+      const {position, expandMode} = item.item;
+      // Trying to activate an already active item
+      if (this.isItemActive(itemId)) return;
+      // Switch between items if currently there is an active one (without collapsing / expanding PS)
+      if (this.activePanels[position] !== null) {
+        this.deactivateItem(this.activePanels[position]!.id);
+      }
+      // Deactivate the opposite panel if is active
+      const oppositePosition: SidePanelPosition = SidePanelsManager.getOppositePanelPosition(position);
+      if (this.activePanels[oppositePosition]) {
+        this.deactivateItem(this.activePanels[oppositePosition]!.id);
+      }
+      // Update new item as active
+      item.componentRef.current?.toggle();
+      this.expand(position, expandMode);
+      this.activePanels[position] = item;
+      item.item.hooks?.onActivate?.();
+    }
+  }
+
+  public deactivateItem(itemId: number): void {
+    const item: ItemMetadata | undefined = this.componentsRegistry.get(itemId);
+    if (item) {
+      const {position} = item.item;
+      this.activePanels[position]?.componentRef.current?.toggle();
+      this.collapse(position);
+      this.activePanels[position] = null;
+      item.item.hooks?.onDeactivate?.();
+    }
+  }
+
+  public isItemActive(itemId: number): boolean {
+    const item: ItemMetadata | undefined = this.componentsRegistry.get(itemId);
+    return item ? this.activePanels[item.item.position]?.id === itemId : false;
+  }
+
+  private toggle(itemId: number): void {
+    if (this.isItemActive(itemId)) {
+      this.deactivateItem(itemId);
+    } else {
+      this.activateItem(itemId);
+    }
+  }
+
+  private expand(position: SidePanelPosition, expandMode: string): void {
+    this.player.ui._uiManager.store.dispatch(ui.reducers.shell.actions.updateSidePanelMode(position, expandMode));
+  }
+
+  private collapse(position: string): void {
+    this.player.ui._uiManager.store.dispatch(
+      ui.reducers.shell.actions.updateSidePanelMode(position, SidePanelModes.HIDDEN)
+    );
+  }
+
+  private injectIconComponent(panelItemData: ItemMetadata): void {
+    const {id, item} = panelItemData;
+    const IconComponent: ComponentClass | FunctionalComponent = item.renderIcon!;
+    const togglePanelFunc: () => void = () => this.toggle(id);
+    this.player.ui.addComponent({
+      label: `Side-Panel-Icon-${item.label}`,
+      presets: [ReservedPresetNames.Playback, ReservedPresetNames.Live],
+      area: ReservedPresetAreas.TopBarRightControls,
+      get: function MyComponent() {
+        return (
+          <div onClick={togglePanelFunc}>
```

Comment:
Does the click continue to the icon component by default? if not need to pass it